File: 프로그래머스/2/42578. 의상/의상.java
// map에 종류별로 배열로 넣고
// +1 개해서 마지막에 -1(아예 안입는 경우)

import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        
        HashMap<String, Integer> mapClothes = new HashMap<String, Integer>();
        for (int i = 0; i < clothes.length ; i++) {
            String type = clothes[i][1];
            
            mapClothes.put(type, mapClothes.getOrDefault(type, 0) + 1);
        }
        
        int answer = 1;
        for (int value : mapClothes.values()) {
            answer *= (value+1);
        }
        
        return answer - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
# 경우의 수
from collections import defaultdict

def solution(clothes):
    answer = 1
    cs = defaultdict(int)
    
    for c, t in clothes:
        cs[t] += 1
    
    for key in cs.keys():
        answer *= (cs[key] + 1)
    
    return answer - 1